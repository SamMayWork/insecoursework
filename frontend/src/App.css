.App {
  position: relative;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media only screen and (min-width: 600px) {
  .navbar {
    overflow-x: hidden;
  }
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.posts {
  display: flex;
  flex-direction: column;
  padding: 1rem;
}

.post {
  display: flex;
  flex-direction: column;
	width: 100%;
	border: 1px solid lightgrey;
	border-radius: 10px;
	padding: 1rem;
	box-sizing: border-box;
	background-color: #FFF;
	margin-bottom: 1rem;
	box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
	transition: all 0.3s cubic-bezier(.25,.8,.25,1);
}

.post_info {
  display: flex;
  flex-direction: row;
}

.post:hover{
	box-shadow: 0 7px 14px rgba(0,0,0,0.25), 0 1px 2px rgba(0,0,0,0.22);
}

.post_date {
  margin-top: auto;
  font-size: 0.9rem;
}

.post_author {
	font-size: 0.9rem;
	margin-bottom: 0.75rem;
	color: rgba(0, 0, 0, 0.5);
}

.post_title {
	font-size: 1.3rem;
	font-weight: 600;
	margin-bottom: 0.5rem;
}

.post_replies {
	font-size: 0.9rem;
	cursor: pointer;
}

.post_replies:hover {
	text-decoration: underline;
}

.post_summary {
	display: flex;
	flex-direction: column;
	flex-basis: 1;
	flex-grow: 1;
}

.post_stats {
	display: flex;
	flex-direction: column;
  flex-grow: 0;
  flex-shrink: 1;
	text-align: right;
}

.post_likes {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
}

.post_like {
  margin-right: 0.5rem;
  box-sizing: border-box;
}

.navbar {
  top: 0;
  left: 0;
  position: sticky;
  display: flex;
  flex-direction: row;
  min-width: 100%;
  background-color: rgba(0, 0, 0, 1);
  overflow-x: scroll;
  box-sizing: border-box;
  height: 3rem;
  overflow-y: hidden;
}

.navbar_link {
  padding: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
}

.navbar_link_anchor {
  display: block;
  color: rgba(255, 255, 255, 0.5);
  text-decoration: none;
  font-weight: 600;
  padding-bottom: 0.2rem
}

.navbar_link_active {
  color: #FFF;
  font-weight: bold;
}

.navbar_link_line {
  display: flex;
  margin-left: auto;
  margin-right: auto;
  height: 2px;
  width: 2rem;
  border-top: 2px solid rgb(0, 0, 196);
}

.comments {
  display: flex;
  flex-direction: column;
  padding: 1rem;
}