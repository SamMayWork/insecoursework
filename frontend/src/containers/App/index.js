/**
 * 
 * App
 * 
 * This component is the skeleton around the actual pages and only contains code
 * which is seen on all pages
 */

import React from 'react';
import ReactDOM from 'react-dom';

import Login from '../LoginPage';
import Dashboard from '../DashboardPage';

import {
  BrowserRouter as Router,
  Switch,
  Route
} from 'react-router-dom';

export default function App() {
  return (
    <Router>
      <Switch>
        <Route exact path="/" component={Login}/>
        <Route path="/dashboard" component={Dashboard}/>
      </Switch>
    </Router>
  );
}