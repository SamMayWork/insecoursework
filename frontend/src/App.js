import React, {Component} from 'react';
import './App.css';
import Post from "./components/Post";
import Posts from "./components/Posts";
import Navbar from "./components/Navbar";
import NavbarLink from "./components/NavbarLink";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      isLoaded: false,
      items: []
    }
  }
  render() {
    const {error, isLoaded, posts } = this.state;
    if (error) {
      return <div>Error: {error.message}</div>
    } else if (!isLoaded) {
      return <div>Loading...</div>
    } else {
      console.log(posts);
      return (
        <div className = "App">
          <Navbar>
            <NavbarLink active>Newest</NavbarLink>
            <NavbarLink>Featured</NavbarLink>
            <NavbarLink>Frequent</NavbarLink>
            <NavbarLink>Recent</NavbarLink>
          </Navbar>
          <Posts>
            {posts.map((post, i) => (
              <Post
                key = {i}
                title = {post.title}
                author = {post.author}
                replies = {post.replies}
                date = {post.date}
              />
            ))}
          </Posts>
        </div>
      );
    }
  }
  componentDidMount() {
    fetch("https://www.reddit.com/r/popular.json")
      .then(res => res.json())
      .then(
        (result) => {
          let posts = [];
          result.data.children.forEach((post) => {
            let postData = post.data;
            let postObj = {
              title: postData.title,
              author: postData.author,
              replies: postData.num_comments,
              date: postData.created * 1000
            };
            posts.push(postObj);
          })
          this.setState({
            isLoaded: true,
            posts: posts
          });
          console.log(result);
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          this.setState({
            isLoaded: true,
            error
          });
        }
      )
  }
}

export default App;