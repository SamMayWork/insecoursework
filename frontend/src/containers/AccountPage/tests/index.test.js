import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import App from '../index';

const renderer = new ShallowRenderer();

describe('<AccountPage />', () => {
	describe('Validation tests', () => {
		it('should render and match the snapshot', () => {
		  renderer.render(<App />);
		  const renderedOutput = renderer.getRenderOutput();
		  expect(renderedOutput).toMatchSnapshot();
		});
	});
});
