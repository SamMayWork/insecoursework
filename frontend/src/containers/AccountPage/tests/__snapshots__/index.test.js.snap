// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountPage /> Validation tests should render and match the snapshot 1`] = `
<div
  className="accountPage"
>
  <BackBar
    title="Account"
  />
  <NotificationZone />
  <PrivacyZone />
  <DangerZone />
</div>
`;
